### PR TITLE
add Haxe support

### PR DESCRIPTION
**Pull request by grepsuzette**
This adds support for the [Haxe](https://haxe.org) language,
thanks to the [ihx](https://lib.haxe.org/p/ihx/) interpreter.

(It has some display glitches here and there sometimes when typing,
but I have been using it for years.)

### CODE REVIEW

File: doc/codi.txt [hunk 1]
```diff
@@ -473,6 +473,12 @@ bin: mathjs
 notes:
   - Evaluation may not work until a newline is entered
 
+haxe~
+bin: ihx 
+maintainer: @grepsuzette
+notes:
+  - `ihx` is an haxe interpreter and should be installed with `haxelib install ihx`
```
Comment (metakirby5) [inline doc/codi.txt:480]: Nit: I think it should be `a Haxe` instead of `an haxe`.

File: doc/codi.txt [hunk 2]
```diff
@@ -473,6 +473,12 @@ bin: mathjs
 notes:
   - Evaluation may not work until a newline is entered
 
+haxe~
+bin: ihx 
+maintainer: @grepsuzette
+notes:
```
Comment (metakirby5) [inline doc/codi.txt:479]: >(It has some display glitches here and there sometimes when typing, but I have been using it for years.)

Do you mind adding a note about that here?

**Review by metakirby5 — CHANGES_REQUESTED** [review]
Thanks for your contribution! Just a few small comments.

File: doc/codi.txt (hunk 2 above)
Comment (grepsuzette) [reply doc/codi.txt:479]: While using codi with python I see the same glitches, so I think we can ignore this :)

**Comment by metakirby5** [comment]
Thank you!

**Pull request merged**